app.js

```diff
@@ -13,76 +13,77 @@ var { pamAuthenticate, pamErrors } = require('node-linux-pam');
 var CUSTOM_PORT = process.env.CUSTOM_PORT || 8888;
 var baserouter = express.Router();
 
-//// JupyterHub auth related code
-const session = require('express-session');
-const passport = require('passport');
-const OAuth2Strategy = require('passport-oauth2');
-const axios = require('axios')
-
-const clientHubApiUrl = process.env.JUPYTERHUB_CLIENT_API_URL
-const serverHubApiUrl = process.env.JUPYTERHUB_API_URL
-const secret = process.env.JUPYTERHUB_API_TOKEN
-const user = process.env.JUPYTERHUB_USER
-
-const passportOptions = {
-  authorizationURL: `${clientHubApiUrl}/oauth2/authorize`,
-  tokenURL: `${serverHubApiUrl}/oauth2/token`,
-  clientID: process.env.JUPYTERHUB_CLIENT_ID,
-  clientSecret: secret
-}
-
-passport.use(new OAuth2Strategy(
-  passportOptions,
-  function(accessToken, refreshToken, params, profile, done) {
-    axios.get(`${serverHubApiUrl}/user`, {
-      headers: { 'Authorization': `Bearer ${params['access_token']}`}
-    }).then(response => {
-      if (!response?.data?.name) {
-        done('Cannot info for loggedin user to authorize access.', null);
-      } else if (response?.data?.name !== user) {
-        done('Logged in user does not match the container\'s user', null);
-      } else {
-        done(null, response.data);
-      }
-    }).catch(error => done(error, null))
+if (!process.env.DISABLE_OAUTH) {
```

Comment:
This solution is perfect but I think that instead of introducing a new environmental variable we could use one of the existing ones especially since this authentication process works only with jupyterhub.

```suggestion
if (process.env.JUPYTERHUB_API_URL) {
```
Note: this also means that you can use the image out-of-the-box.